src/Razor/src/Microsoft.AspNetCore.Razor.Language/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -305,14 +291,24 @@ public static bool IsSelfClosing(this MarkupTagBlockSyntax tagBlock)
             return lastChild?.GetContent().EndsWith("/>", StringComparison.Ordinal) ?? false;
         }
 
-        public static bool IsVoidElement(this MarkupTagBlockSyntax tagBlock)
+        public static bool IsVoidElement(this MarkupStartTagSyntax startTag)
         {
-            if (tagBlock == null)
+            if (startTag == null)
             {
-                throw new ArgumentNullException(nameof(tagBlock));
+                throw new ArgumentNullException(nameof(startTag));
+            }
+
+            return ParserHelpers.VoidElements.Contains(startTag.GetTagName());
+        }
+
+        public static bool IsVoidElement(this MarkupEndTagSyntax endTag)
```

Comment:
Why do we care if the end tag is a void element? Can't we just treat all end tags equivalently